mk/spksrc.archs.mk

```diff
@@ -0,0 +1,36 @@
+# Common arch definitions
+
+# All available CPU architectures
+
+# Distinct SRM and DSM archs to allow handling of different TCVERSION ranges.
+# SRM - Synology Router Manager
+SRM_ARM7_ARCHS = northstarplus ipq806x dakota
+# DSM - all ARMv7 except SRM specific archs
+DSM_ARM7_ARCHS = alpine armada370 armada375 armada38x armadaxp comcerto2k monaco
+
+# Generic archs used for packages supporting multiple archs (where applicable)
+GENERIC_ARMv7_ARCH = armv7
+GENERIC_ARM64_ARCH = aarch64
+GENERIC_x64_ARCH = x64
+GENERIC_ARCHS = $(GENERIC_ARMv7_ARCH) $(GENERIC_ARM64_ARCH) $(GENERIC_x64_ARCH)
+
+ARM5_ARCHS = 88f6281
+ARM7_ARCHS = $(GENERIC_ARMv7_ARCH) $(DSM_ARM7_ARCHS) $(SRM_ARM7_ARCHS)
+# hi3535 is not supported by generic ARMv7 arch
+ARM7L_ARCHS = hi3535
+ARM8_ARCHS = $(GENERIC_ARM64_ARCH) rtd1296 armada37xx
+ARM_ARCHS = $(ARM5_ARCHS) $(ARM7_ARCHS) $(ARM7L_ARCHS) $(ARM8_ARCHS)
```

Comment:
I'm having a little trouble here with the ARM naming overall.
ARM7 family and ARMv7 architecture does not have the same meaning at all but it becomes really hard to follow.
Does this needs to be addressed now or within another PR?  Duno.
But at least it should be much easier as now separated into its own file which is a great idea.
See: https://en.wikipedia.org/wiki/List_of_ARM_microarchitectures

Reply:
`armv7` (all lowercase) is the generic arch for ARM7 architectures in the same way as `x64` is for x86_64 and `aarch64` for ARM8.
These generic archs are pseudonyms to distinguish from real archs.
Under the hood there are no dedicated toolchains for these. But the toolchains of the generic archs contain all the dedicated archs that are compatible (variable TC_ARCH) that will define the "arch" variable in the INFO file of the generict spk packages.
The mapping used for generic toolchains is
- x64 -> pineview
- armv7 -> armada38x
- aarch64 -> rtd1296

I hope this helps. I really wanted to clarify such definitions with this PR.

Reply:
I have no issue with that beside that in reality `ARM7` family actually includes arm v3-4-5 architectures.
So the usage of ARM with no `V` isn't appropriate overall as it describe something else entirely.
Personally, I'd rather rename all `ARMX` to using a **V** such as `ARMV7_ARCHS`

Reply:
Finally renamed to match your preference 😃